record/ltsv.go

```diff
@@ -1,63 +1,61 @@
 package record
 
 import (
+	"bufio"
+	"io"
 	"strconv"
-	"strings"
-
-	"github.com/reproio/columnify/schema"
 
 	"github.com/Songmu/go-ltsv"
 )
 
-func FormatLtsvToMap(data []byte) ([]map[string]interface{}, error) {
-	lines := strings.Split(string(data), "\n")
+type ltsvInnerDecoder struct {
+	s *bufio.Scanner
+}
+
+func newLtsvInnerDecoder(r io.Reader) *ltsvInnerDecoder {
+	return &ltsvInnerDecoder{
+		s: bufio.NewScanner(r),
+	}
+}
 
-	records := make([]map[string]interface{}, 0)
-	for _, l := range lines {
-		v := map[string]string{}
+func (d *ltsvInnerDecoder) Decode(r *map[string]interface{}) error {
+	if d.s.Scan() {
+		data := d.s.Bytes()
 
-		err := ltsv.Unmarshal([]byte(l), &v)
+		m := map[string]string{}
+		err := ltsv.Unmarshal(data, &m)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		m := make(map[string]interface{})
-		for k, v := range v {
+		*r = make(map[string]interface{})
+		for k, v := range m {
 			// bool
 			if v != "0" && v != "1" {
 				if vv, err := strconv.ParseBool(v); err == nil {
```

Comment:
Why don't you use the schema information? I'm concerned that the string `"t"` is converted `true`.

Reply:
same as https://github.com/reproio/columnify/pull/52#discussion_r471116447